packages/app/src/explorer/utils.ts

```diff
@@ -26,3 +28,52 @@ export function needsNxBadge(
 
   return false;
 }
+
+function getExplorerButtonList(): HTMLButtonElement[] {
+  const explorer = document.querySelector(`#${EXPLORER_ID}`);
+  if (!explorer) {
+    return [];
+  }
+  return [...explorer.querySelectorAll('button')];
+}
+
+export function focusParent(e: { preventDefault: () => void }, path: string) {
+  const buttonList = getExplorerButtonList();
+  const parentPath = path.slice(0, path.lastIndexOf('/')) || '/';
+
+  const parentButton = buttonList.find(
+    (element) => element.getAttribute('aria-label') === parentPath
+  );
+  if (parentButton) {
+    parentButton.focus();
+    e.preventDefault();
+  }
+}
+
+export function focusNext(e: { preventDefault: () => void }) {
+  const { activeElement } = document;
+  if (!(activeElement instanceof HTMLButtonElement)) {
```

Comment:
I choose to no longer check here if `activeElement` is part of `Explorer` to be able to encapuslate the ref to `Explorer` in `getExplorerButtonList`.

Anyway, if it is not part of `Explorer`, `activeIndex` below will be set to `-1` and the focus will not change, giving the same result as before.

Reply:
Yep, makes sense, and if the button receives the `keydown` event, it means it's focused anyway, so you could just use `evt.currentTarget`.